lib/rules/require-super-in-lifecycle-hooks.js

```diff
@@ -44,7 +44,8 @@ function isNativeSuper(node, hook) {
     types.isMemberExpression(node.callee) &&
     node.callee.object.type === 'Super' &&
     types.isIdentifier(node.callee.property) &&
-    node.callee.property.name === hook
+    node.callee.property.name === hook &&
+    (hook !== 'init' || node.arguments.length > 0)
```

Comment:
This probably shouldn't be inside `isNativeSuper()` since the presence of arguments doesn't affect whether it's a native super call.

Also, the arguments check should probably be more specific than just checking the count.

Reply:
Absolutely agree. I have moved the check under `checkAndReport` function: There it checks if the init hook already hasSuper and if so, does that super have arguments.

Though, the argument check still looks at the count. I am not sure how I can make it more specific 🤔

Reply:
Okay, I think you're right that the arguments length check is sufficient, to allow flexibility for however somebody wants to pass the arguments.